app/src/main/kotlin/com/simplemobiletools/flashlight/activities/WidgetTorchConfigureActivity.kt

```diff
@@ -69,67 +79,31 @@ class WidgetTorchConfigureActivity : SimpleActivity() {
         }
     }
 
-    private fun initVariables() {
-        mWidgetColor = config.widgetBgColor
-        if (mWidgetColor == resources.getColor(R.color.default_widget_bg_color) && config.isUsingSystemTheme) {
-            mWidgetColor = resources.getColor(R.color.you_primary_color, theme)
-        }
-
-        mWidgetAlpha = Color.alpha(mWidgetColor) / 255.toFloat()
-
-        mWidgetColorWithoutTransparency = Color.rgb(Color.red(mWidgetColor), Color.green(mWidgetColor), Color.blue(mWidgetColor))
-        binding.configWidgetSeekbar.apply {
-            setOnSeekBarChangeListener(seekbarChangeListener)
-            progress = (mWidgetAlpha * 100).toInt()
-        }
-        updateColors()
-    }
-
     private fun saveConfig() {
-        config.widgetBgColor = mWidgetColor
+        config.widgetBgColor = viewModel.widgetColor.value
         requestWidgetUpdate()
 
         Intent().apply {
-            putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, mWidgetId)
+            putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, viewModel.widgetId.value)
             setResult(Activity.RESULT_OK, this)
         }
         finish()
     }
 
     private fun pickBackgroundColor() {
-        ColorPickerDialog(this, mWidgetColorWithoutTransparency) { wasPositivePressed, color ->
+        ColorPickerDialog(this, viewModel.widgetColor.value) { wasPositivePressed, color ->
```

Comment:
Before we merge the PR we should have this one migrated to Compose, otherwise it doesn't make sense.